mailpoet/lib/Newsletter/NewslettersRepository.php

```diff
@@ -484,6 +492,27 @@ public function bulkDelete(array $ids) {
         ->where('n.id IN (:ids)')
         ->setParameter('ids', $ids)
         ->getQuery()->execute();
+
+
+      $entityTypesToBeDetached = [
+        StatisticsNewsletterEntity::class,
+        StatisticsOpenEntity::class,
+        StatisticsClickEntity::class,
+        NewsletterPostEntity::class,
+        NewsletterOptionEntity::class,
+        NewsletterLinkEntity::class,
+        StatsNotificationEntity::class,
+        SendingQueueEntity::class,
+        ScheduledTaskSubscriberEntity::class,
+        NewsletterSegmentEntity::class,
+      ];
+      foreach ($entityTypesToBeDetached as $entityType) {
+        $this->detachEntitiesOfType($entityType, $isRelatedNewsletterToBeDeleted);
+      }
+
+      $this->detachEntitiesOfType(ScheduledTaskEntity::class, function($entity) use ($taskIds): bool {
```

Comment:
I'm checking this deeper and the `$taskIds` seems to come from stats notification tasks (https://github.com/mailpoet/mailpoet/pull/5374/files#diff-365816efcb88f989e0039dfbd79719a9d5ec37961bae01b7744f25a674d907b8R430), which might not be what's expected (= all task IDs related to the deleted newsletters).